lib/resty/core/uri.lua

```diff
@@ -53,22 +53,34 @@ elseif subsystem == "stream" then
 end
 
 
-ngx.escape_uri = function (s)
+ngx.escape_uri = function (s, esc_type)
     if type(s) ~= 'string' then
         if not s then
             s = ''
         else
             s = tostring(s)
         end
     end
+
+    if esc_type == nil then
+        esc_type = ngx.ESCAPE_URI_COMPONENT
+    else
+        if type(esc_type) ~= 'number' then
+            error("\"esc_type\" is not number", 3)
+        end
+        if esc_type < ngx.ESCAPE_URI or esc_type > ngx.ESCAPE_MAIL_AUTH then
+            error("\"esc_type\" " .. esc_type .. " out of range", 3)
```

Comment:
Please do not use your internal identifier in the error message. The end user won't have a clue unless they look at the actual source code.